torch/csrc/jit/passes/check_strict_fusion.cpp

```diff
@@ -0,0 +1,123 @@
+
+#include <torch/csrc/jit/passes/check_strict_fusion.h>
+
+#include <c10/util/Exception.h>
+#include <torch/csrc/jit/frontend/error_report.h>
+#include <torch/csrc/jit/ir/ir.h>
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/quantization/helper.h>
+#include <torch/csrc/jit/runtime/graph_iterator.h>
+#include <unordered_map>
+
+namespace torch {
+namespace jit {
+
+namespace {
+
+bool isStrictFusion(Value* value) {
+  const auto class_name = getModuleName(value);
+  return class_name.has_value() &&
+      (*class_name == "__torch__.torch.jit.strict_fusion");
+}
+
+} // namespace
+
+bool fusionGuardCheck(Symbol k) {
+  return k == Symbol::prim("TensorExprDynamicGuard") || k == prim::TypeCheck ||
+      k == prim::CudaFusionGuard || k == prim::RequiresGradCheck;
```

Comment:
do you know if there's a way to put this into the fusers themselves? e.g. if/when new fusers get added, they would need to know to add their guards here. (you don't need to change this, just wondering)

Reply:
Can do as follow up